pages/item/[id].js

```diff
@@ -27,23 +27,56 @@ const StyledItemDetails = styled.div`
   }
 `;
 
+const ItemDetailContainer = styled.div`
+  max-width: 600px;
+  margin: 20px auto;
+  padding: 20px;
+  border: 1px solid #ccc;
+  border-radius: 8px;
+`;
+const StyledLink = styled(Link)`
+  display: inline-block;
+  margin-bottom: 20px;
+  text-decoration: none;
+  color: #007bff;
+  &:hover {
+    text-decoration: underline;
+  }
+`;
+
 const ItemDetails = () => {
   const router = useRouter();
   const { id } = router.query;
-  console.log(id);
-  const item = initialItems.find((item) => item.id === id);
+  const [item, setItem] = useState(null);
+
+  useEffect(() => {
+    if (id) {
+      const storedItems =
+        JSON.parse(localStorage.getItem("shoppingItems")) || [];
+      const foundItem = storedItems.find((item) => item.id === id);
+
+      if (foundItem) {
+        setItem(foundItem);
+      } else {
+        setItem(null);
+      }
+    }
+  }, [id]);
```

Comment:
**As mentioned above:** 🙃
Instead of manually using localStorage.getItem and localStorage.setItem, please useLocalStorageState to persist your items state.

This will make the code cleaner and easier to maintain, as shown in the course. ✨

🔗 Here is a link to the [handout](https://github.com/neuefische/hh-web-24-7/blob/main/sessions/react-with-local-storage/react-with-local-storage.md#react-with-local-storage-1)